sinks.go

```diff
@@ -144,6 +144,35 @@ func ToChannel[T any](ctx context.Context, src Observable[T], opts ...ToChannelO
 	return items
 }
 
+// ToAsyncChannel is like ToChannel but with a local buffer to decouple the
+// source observable from the observer.
+// It is useful when the source observable cannot be delayed by a slow consumer
+// and it is safe for the consumer to lose intermediate items while busy.
+func ToAsyncChannel[T any](ctx context.Context, src Observable[T], opts ...ToChannelOpt) <-chan T {
```

Comment:
The implementation looks good to me, but I'm not entirely sure about the name. `ToLossyChannel` might be an option, but don't quite like that either. `ToTruncatingChannel`?